src/components/node-list/node-list-row.js

```diff
@@ -163,6 +163,11 @@ const NodeListRow = memo(
           <input
             id={id}
             className="pipeline-nodelist__row__checkbox"
+            data-heap-event={
+              kind === 'element'
+                ? `focusMode.checked.${checked}`
+                : `visible.${name}.${checked}`
```

Comment:
given that we are expecting different heap events to be sent for different node types ( `dataset`, `Parameters`, `Nodes`), we would need to add in the relevant test to ensure the right heap event gets assigned for each node type.